src/components/toggle/toggle-option/toggle-option.ts

```diff
@@ -81,28 +68,19 @@ export class SbbToggleOptionElement extends SbbIconNameMixin(LitElement) {
     super.willUpdate(changedProperties);
 
     if (changedProperties.has('checked')) {
-      this._handleCheckedChange(this.checked, changedProperties.get('checked')!);
-    }
-    if (changedProperties.has('value')) {
-      this._handleValueChange(this.value, changedProperties.get('value')!);
+      this.setAttribute('aria-checked', `${this.checked}`);
+      this._verifyTabindex();
+      if (this.checked) {
+        this._syncOptions();
+      }
     }
     if (changedProperties.has('disabled')) {
       this._handleDisabledChange();
     }
   }
 
-  private _handleCheckedChange(currentValue: boolean, previousValue: boolean): void {
-    if (currentValue !== previousValue) {
-      this.setAttribute('aria-checked', `${this.checked}`);
-      this._stateChange.emit({ type: 'checked', checked: this.checked });
-      this._verifyTabindex();
-    }
-  }
-
-  private _handleValueChange(currentValue: string | null, previousValue: string | null): void {
-    if (this.checked && currentValue !== previousValue) {
-      this._stateChange.emit({ type: 'value', value: currentValue });
-    }
+  private _syncOptions(): void {
```

Comment:
Nitpick; While correct, I feel like the name is not precise enough.
Maybe `_uncheckOtherOptions`?